arm-devtestlabs/2016-05-15/swagger/DTL.json

```diff
@@ -2486,346 +2476,4367 @@
+            }
+        },
+        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DevTestLab/labs/{labName}/virtualmachines/{name}/start": {
+            "post": {
+                "tags": [ "VirtualMachine" ],
+                "description": "Start a virtual machine. This operation can take a while to complete.",
+                "operationId": "VirtualMachine_Start",
+                "parameters": [
+                    { "$ref": "#/parameters/subscriptionId" },
+                    { "$ref": "#/parameters/resourceGroupName" },
+                    {
+                        "name": "labName",
+                        "in": "path",
+                        "description": "The name of the lab.",
+                        "required": true,
+                        "type": "string"
+                    },
+                    {
+                        "name": "name",
+                        "in": "path",
+                        "description": "The name of the virtual machine.",
+                        "required": true,
+                        "type": "string"
+                    },
+                    { "$ref": "#/parameters/api-version" }
+                ],
+                "responses": {
+                    "200": { "description": "OK" },
+                    "202": { "description": "Accepted" },
+                    "default": {
+                        "description": "BadRequest",
+                        "schema": { "$ref": "#/definitions/CloudError" }
+                    }
+                },
+                "security": [ { "oauth2": [ "user_impersonation" ] } ],
+                "x-ms-long-running-operation": true
+            }
+        },
+        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DevTestLab/labs/{labName}/virtualmachines/{name}/stop": {
+            "post": {
+                "tags": [ "VirtualMachine" ],
+                "description": "Stop a virtual machine This operation can take a while to complete.",
+                "operationId": "VirtualMachine_Stop",
+                "parameters": [
+                    { "$ref": "#/parameters/subscriptionId" },
+                    { "$ref": "#/parameters/resourceGroupName" },
+                    {
+                        "name": "labName",
+                        "in": "path",
+                        "description": "The name of the lab.",
+                        "required": true,
+                        "type": "string"
+                    },
+                    {
+                        "name": "name",
+                        "in": "path",
+                        "description": "The name of the virtual machine.",
+                        "required": true,
+                        "type": "string"
+                    },
+                    { "$ref": "#/parameters/api-version" }
+                ],
+                "responses": {
+ 
```

Comment:
As you mentioned that `name` & `labName` both are required properties. That is right. [name](https://github.com/Azure/azure-rest-api-specs/pull/1067/files/98475064dccdacddc1ee7b4c8b120de790703389#diff-11fa414b99a450d8e3f7a81da88e39ffR4430) - Name of the lab is defined here but I do not see `name` - Name of the vm in the model [LabVirtualMachineCreationParameter](https://github.com/Azure/azure-rest-api-specs/pull/1067/files/98475064dccdacddc1ee7b4c8b120de790703389#diff-11fa414b99a450d8e3f7a81da88e39ffR6548). Which inherits `name` from `Resource` model by using [allOf](https://github.com/Azure/azure-rest-api-specs/pull/1067/files/98475064dccdacddc1ee7b4c8b120de790703389#diff-11fa414b99a450d8e3f7a81da88e39ffR6552) but that is [readOnly](https://github.com/Azure/azure-rest-api-specs/pull/1067/files/98475064dccdacddc1ee7b4c8b120de790703389#diff-11fa414b99a450d8e3f7a81da88e39ffR7492) `name` not the one that we can send over wire. 

Does that clarify my question?